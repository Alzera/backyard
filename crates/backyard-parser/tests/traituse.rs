use backyard_parser::parse_eval;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "class A { use Ale; }").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn items() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(
    &arena,
    "class A {
  use Loggable, Usable {
    log as public;
    get as private alias;
    Loggable::log as aliasLoggable;
    Usable insteadof Loggable;
    Usable::useResource insteadof Loggable;
  }
}"
  ).unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}
