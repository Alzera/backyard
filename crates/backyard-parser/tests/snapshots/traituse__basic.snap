---
source: crates/backyard-parser/tests/traituse.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: class
    inheritance: ~
    name:
      node_type: identifier
      name:
        - 65
      loc:
        start:
          line: 1
          column: 6
          offset: 6
        end:
          line: 1
          column: 7
          offset: 7
      leadings: ~
      trailings: ~
    extends: ~
    implements: []
    body:
      node_type: block
      statements:
        - node_type: trait_use
          traits:
            - node_type: identifier
              name:
                - 65
                - 108
                - 101
              loc:
                start:
                  line: 1
                  column: 14
                  offset: 14
                end:
                  line: 1
                  column: 17
                  offset: 17
              leadings: ~
              trailings: ~
          adaptations: []
          loc:
            start:
              line: 1
              column: 10
              offset: 10
            end:
              line: 1
              column: 14
              offset: 14
          leadings: ~
          trailings: ~
      loc:
        start:
          line: 1
          column: 8
          offset: 8
        end:
          line: 1
          column: 19
          offset: 19
      leadings: ~
      trailings: ~
    is_readonly: false
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 1
        column: 19
        offset: 19
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 19
    offset: 19
leadings: ~
trailings: ~
